doc/ug/getting_started_sw_testing.md

```diff
@@ -0,0 +1,100 @@
+
+The OpenTitan software tests fall into two main categories:
+
+* Unit tests
+* Smoke tests
+
+Unit tests are designed to run on the host so they are quick to build and run.
+Writing unit tests is a good way to provide high levels of test coverage for a library without significantly impacting the runtime of the continuous integration tests.
+Smoke tests on the other hand are designed to run on an emulated device and therefore require a more complicated setup procedure.
+Typically smoke tests take significantly longer to run than unit tests.
+Smoke tests are therefore best suited for testing that the core functionality of a library is correct.
+
+## Unit tests
+
+OpenTitan unit tests are written using the [GoogleTest](https://github.com/google/googletest/) C++ test framework and are designed to be run on a workstation.
+Typically this means that they are executed in a Linux environment running on the x86_64 architecture although they should be designed to be as portable as possible.
+Device-specific features that are not necessarily available on the host system, for example memory-mapped I/O (MMIO) and Ibex control registers (CSRs), may be provided by mock implementations.
+
+Unit tests use the suffix `_unittest.cc` and are typically contained in the same directory as other tests for the same code.
+
+The unit tests may be executed by invoking the ninja test target:
+
+```console
+# Configure the Meson environment.
+$ cd "$REPO_TOP"
+$ ./meson_init.sh
+
+# Build and run the unit tests.
+$ ninja -C build-out test
+```
+
+## Device smoke tests
+
+OpenTitan tests that are designed to be executed on the target device itself are referred to as ‘smoke tests’.
+Smoke tests are full applications written in a language suitable for the target device such as C.
+The target device may be emulated using either software simulation or a programmable hardware device (FPGA).
+
+The name ‘smoke test’ is used to describe device tests because they are not typically as comprehensive as unit tests that target the host.
+This is because device emulation, especially when using a software simulation, is relatively slow.
+Therefore the execution time of smoke tests needs to be kept as low as possible.
+
+Smoke tests use the suffix `_smoketest.c` and are usually found in the same directory as other tests for the same code.
+
+To build a software simulation of the target device run the following command:
+
+<div class="bd-callout bd-callout-warning">
+  <h5>Note</h5>
+
+  The software simulation needs to be rebuilt every time the source code for the hardware changes.
+  The `fusesoc` command should therefore be re-run every time a branch is checked out or a git pull/rebase operation is performed.
+
+  Failure to rebuild the software simulator may result in the tests failing.
+</div>
+
+```console
+# Build Verilator simulation.
+$ cd "$REPO_TOP"
+$ fusesoc --cores-root . run --flag=fileset_top --target=sim --setup --build lowrisc:systems:top_earlgrey_verilator
+```
```

Comment:
The alternative to building it on your own is downloading the CI-built distribution artifacts.

We should really add a simple script to get the most recent artifacts from CI, but for now the manual approach is:

* https://dev.azure.com/lowrisc/opentitan/_build?definitionId=9&_a=summary&repositoryFilter=5&branchFilter=566%2C566%2C566%2C566%2C566
* Choose the latest build
* Go to "7 published" on the top.
* Download the opentitan-dist artifact.

If you need that more often, maybe you can look into writing such a script using a bit of Azure API?